feat(compliance): add CCC catalogs for AWS, Azure and GCP

pedrooot opened the pull request:
### Description

This PR adds CCC-Objects mapping with Prowler compliance framework format.
<img width="412" height="151" alt="Screenshot 2025-10-14 at 17 51 38" src="https://github.com/user-attachments/assets/6b2dc5c0-f725-425d-a001-0b581ff52970" />

Still in draft since some changes needs to be addressed.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.
- [x] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### API
- [x] Verify if API specs need to be regenerated.
- [x] Check if version updates are required (e.g., specs, Poetry, etc.).
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/api/CHANGELOG.md), if applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

pedrooot committed:
feat(compliance): add CCC-Objects for aws

pedrooot committed:
feat(ccc): make some tweaks

pedrooot committed:
feat(compliance): improve CCC description

pedrooot committed:
feat(azure): add ccc azure json

pedrooot committed:
feat(azure): add ccc azure json

pedrooot committed:
feat(gcp): add ccc objects

pedrooot committed:
feat(ccc): add more catalogs

pedrooot committed:
feat(ccc): unify all controls in a single file

pedrooot committed:
Merge branch 'master' into PRWLR-7257-map-ccc-objects-in-aws-provider

pedrooot committed:
feat(ccc): add remaining catalogs

pedrooot committed:
feat(api): add export changes

pedrooot committed:
feat(compliance): add Common Cloud Controls for AWS, Azure & GCP

pedrooot committed:
chore(changelog): update with latest changes

alejandrobailo commented on ui/components/compliance/compliance-custom-details/ccc-details.tsx:
```diff
@@ -0,0 +1,146 @@
+    const mappings = requirement.section_guideline_mappings as Array<{
+      ReferenceId: string;
+      Identifiers: string[];
+    }>;
```
Same as above

Alan-TheGentleman committed:
Merge branch 'master' into PRWLR-7257-map-ccc-objects-in-aws-provider

alejandrobailo reviewed (changes requested):
Good work. 
Some changes: just keep mappings in the /lib/compliance files and the details in the react components.

Alan-TheGentleman committed:
refactor(compliance): extract CCC section configuration into constants

- Move CCC text and mapping section definitions to shared constants
- Replace inline logic with configuration-driven approach
- Add TypeScript interfaces for section configurations
- Use REQUIREMENT_STATUS constants instead of hardcoded strings

Alan-TheGentleman committed:
chore(compliance): reorder imports in ccc.tsx

pedrooot committed:
fix(ui): lint

pedrooot committed:
fix(ui): lint

pedrooot committed:
fix: lint

pedrooot commented on ui/components/icons/compliance/IconCompliance.tsx:
```diff
@@ -18,6 +19,9 @@ import RBILogo from "./rbi.svg";
 export const getComplianceIcon = (complianceTitle: string) => {
+  if (complianceTitle.toLowerCase().includes("ccc")) {
+    return CCCLogo;
+  }
```
Changed by @Alan-TheGentleman
